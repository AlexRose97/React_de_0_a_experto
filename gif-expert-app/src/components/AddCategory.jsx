import { useState } from "react";

export const AddCategory = ({ onNewValue }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    if (inputValue.trim().length > 1) {
      onNewValue(inputValue.trim());
      setInputValue("");//reiniciar el input
    }
    event.preventDefault();
  };
  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      ></input>
    </form>
  );
};
