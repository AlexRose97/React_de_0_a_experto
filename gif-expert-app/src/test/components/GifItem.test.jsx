
import {render} from '@testing-library/react'
import { GifItem } from '../../components';



describe('Pruebas en GiftItem', () => {
    const title="Saitama";
    const url="https://Saitama";
    test('debe de hacer match con el snapshot', () => {
       const {container} =render(<GifItem title={title} url={url}/>)
       expect(container).toMatchSnapshot();
    });
});
