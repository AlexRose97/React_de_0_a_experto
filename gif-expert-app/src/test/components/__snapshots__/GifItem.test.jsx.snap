// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama/imagen.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
